storage/local/storage.go

```diff
@@ -286,7 +286,47 @@ func (s *memorySeriesStorage) NewIterator(fp clientmodel.Fingerprint) SeriesIter
 		// return any values.
 		return nopSeriesIterator{}
 	}
-	return series.newIterator()
+	return &boundedIterator{
+		it:    series.newIterator(),
+		start: clientmodel.Now().Add(-s.dropAfter),
```

Comment:
I think the approach is fine in general. It makes me a bit uneasy that you'll have in theory different start times for each series iterator in a set of iterators needed for an evaluation... Since our timestamps are only millisecond precision, that's probably a merely academic concern... Perhaps add that to the doc comment of NewIterator (in interface.go): 

"The iterator will never return samples older than retention time, relative to the time NewIterator was called."